homeassistant/components/lifx/light.py

```diff
@@ -193,13 +193,17 @@ async def async_setup_entry(
     """Set up LIFX from a config entry."""
     # Priority 1: manual config
     if not (interfaces := hass.data[LIFX_DOMAIN].get(DOMAIN)):
-        # Priority 2: Home Assistant enabled interfaces
-        ip_addresses = (
-            source_ip
-            for source_ip in await network.async_get_enabled_source_ips(hass)
-            if isinstance(source_ip, IPv4Address) and not source_ip.is_loopback
-        )
-        interfaces = [{CONF_SERVER: str(ip)} for ip in ip_addresses]
+        # Priority 2: scanned interfaces
+        lifx_ip_addresses = await aiolifx().LifxScan(hass.loop).scan()
+        interfaces = [{CONF_SERVER: ip} for ip in lifx_ip_addresses]
```

Comment:
We should only scan on Home Assistant-enabled interfaces.

Reply:
>  but no-one knows about Home Assistant's network configuration

That is not an integration issue to solve.

Reply:
It's not my problem to solve either, so I vote with restoring the LIFX functionality until someone else decides how to solve it.

Reply:
> It's not my problem to solve either

I'm sorry, I don't mean to be harsh, we are trying to solve issue here. And having discussions about it and following general design rules and principles around that, sounds like a normal thing to do to me.

This is your PR right? It seems like you try to solve a problem, and I'm jumping into the review process. I'm happy to help, discuss and getting things somewhere. But it seems like you are not interested (for whatever reason).

Anyways, tried to find more information for you on the subject: <https://github.com/home-assistant/architecture/discussions/509>

Just in case you wondered about more context.

Reply:
If I'd written the integration in the first place, I'd be committed to fixing it. And the custom integration I'm working on follows the rules.

As this is a built in integration, I think the core developers should fix this. This PR is my recommendation in the interim until that happens.